### PR TITLE
Add a test case to enforce rescaling behavior

### LINKED ISSUE #601: Test case to ensure include_rescaling works
We need to add a test to models_test.py to cover this case?  

This unit test should check that for all models, if include_rescaling=True a Rescaling layer is present in the models, and if it is false a Rescaling layer is NOT true.

This came up in https://github.com/keras-team/keras-cv/pull/558

### PR DESCRIPTION
**Pull request by ianjjohnson**
# What does this PR do?

Adds a new test case for models to verify that rescaling does occur when specified and does not occur when not specified.

Fixes #601

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [x] Did you write any new necessary tests?

## Who can review?
@LukeWood @tanzhenyu

### CODE REVIEW

File: keras_cv/models/models_test.py [hunk 1]
```diff
@@ -283,6 +283,42 @@ def test_application_variable_input_channels(self, app, last_dim):
...
+                    input_shape=(224, 224, 3),
+                    include_top=True,
+                    num_classes=1000,
+                    include_rescaling=include_rescaling,
+                    weights=None,
+                )
+            elif "MixerB16" in app.__name__ or "MixerL16" in app.__name__:
+                model = app(
+                    input_shape=(224, 224, 3),
+                    patch_size=(16, 16),
```
Comment (LukeWood) [inline keras_cv/models/models_test.py:300]: Lets update these tests to remove hard coded checks in the test body.  

Instead, lets introduce support for "kwargs" in MODEL_LIST as an extra parameter.

So for models with no extra params you'll have:

```
    (darknet.DarkNet53, 512, {}),
```

For models that do require more info:

```
    (mlp_mixer.MLPMixerL16, 1024, {"patch_size": (16, 16))
```

Unit tests will take:

```
    def test_rescaling(self, app, args):
```

and model construction will be done:

```
model = app(
                    input_shape=(224, 224, 3),
                    include_top=True,
                    num_classes=1000,
                    include_rescaling=include_rescaling,
                    weights=None,
                    **extra_args
                )
```

**Review by LukeWood — CHANGES_REQUESTED** [review]
readability improvement that will be more important as time goes on, other than that LGTM!

File: keras_cv/models/models_test.py [hunk 2]
```diff
@@ -283,6 +283,42 @@ def test_application_variable_input_channels(self, app, last_dim):
...
+                model = app(
+                    input_shape=(224, 224, 3),
+                    patch_size=(32, 32),
+                    include_top=True,
+                    num_classes=1000,
+                    include_rescaling=include_rescaling,
+                    weights=None,
+                )
+
+            if include_rescaling:
```
Comment (LukeWood) [inline keras_cv/models/models_test.py:316]: lets have one test with & without:

```
test_rescaling_true
```

```
test_rescaling_false
```

to ensure we hit both cases for all models.

**Comment by LukeWood** [comment]
Thanks for the swift PR, some minor changes requested!

File: keras_cv/models/models_test.py (hunk 2 above)
Comment (ianjjohnson) [reply keras_cv/models/models_test.py]: I realized these cases are covered by existing tests, so I moved the assertions into those cases.

File: keras_cv/models/models_test.py (hunk 1 above)
Comment (ianjjohnson) [reply keras_cv/models/models_test.py]: I've adopted the args model, which has cleaned up this code quite a bit. It's still imperfect due to the output_shape checks, but I think this is a big improvement.

**Comment by ianjjohnson** [comment]
Note that I also uncovered that VGG needs to have a specified input_shape due to the dense layer needing an input shape hint. This required a small change to the tests.

**Review by LukeWood — APPROVED** [review]
Thank you Ian!!!

**Pull request merged**